hack/verify-import-aliases.sh

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env bash
+
+# Copyright 2017 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+KUBE_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+source "${KUBE_ROOT}/hack/lib/init.sh"
+
+kube::golang::verify_go_version
+
+cd "${KUBE_ROOT}"
+
+ret=0
+go run cmd/preferredimports/preferredimports.go "$@" || ret=$?
```

Comment:
git diff --name-only master | go run cmd/preferredimports/preferredimports.go "$@" || ret=$?

Reply:
If the idea is to run the script only for the current changes, there's another way to do it already

```
git diff --name-only "$(git merge-base HEAD master)"...HEAD | sed 's|\(.*\)/.*|\1|' | sort | uniq | xargs go run cmd/preferredimports/preferredimports.go
```

Reply:
That will force people to update aliases as they touch existing files. Similar things happen with lint occasionally (the lint tool will start complaining about a file that is passing lint checks on master, and require fixup in an unrelated PR). Failing CI on changes unrelated to a PR is probably friction we don't need.

Reply:
Agree, that's why i was running on the whole code base and not limiting to the current change

Reply:
Does it mean that if we want to add a new rule, we have to fix all of the import aliases in the whole codebase? If yes, then we have to work on a `update-import-aliases` script, which could always help us to fix up the code.

Reply:
Yes